.github/workflows/ci.yml

```diff
@@ -322,3 +335,39 @@ jobs:
           "$(jq -r 'all(.result as $result | (["success", "skipped"] | contains([$result])))' <<< "${needs}")"
         env:
           needs: ${{ toJson(needs) }}
+
+  code-cov:
+    name: Coverage
+    needs: [test, build, conclusion, generate-matrix]
+    # should check that there are any artifacts, if not skip
+    if: always() && (needs.build.result == 'success' || needs.build.result == 'skipped') && needs.test.result == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
+      - uses: actions/download-artifact@v3
+        with:
+          path: ${{ runner.temp }}/artifacts
+      - name: Grab PR number
+        run: echo "::set-output name=pr::"$(echo $commit_message | sed -ne  's/.*\#\(.*\):/\1/p')
+        id: pr-number
+        if: ${{ !github.event.pull_request.number }}
+        env:
+          commit_message: >
+            ${{
+              ((
+                startsWith(github.event.head_commit.message, 'Try #') &&
+                github.event.head_commit.author.username == 'bors[bot]'
+              ) && github.event.head_commit.message) || ''
+            }}
+      - name: Upload to codecov.io
+        run: |
+          set -x
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          chmod +x codecov
+          sha_rev=$(git rev-parse HEAD)
+          find ${artifacts} -name "lcov.*.info" -exec ./codecov -F $(echo {} | sed -n 's/lcov\.\(.*\)\.info/\1/p') \
+            ${pr:+-P ${pr}} -f {} --sha ${sha_rev} -n $(echo {} | sed -n 's/lcov\.\(.*\)\.info/\1/p') \;
```

Comment:
we may have to specify parent sha here, not sure if it really works when you have a staging/trying branch.

been trying to find prior art for codecov with bors integration but not able to find any good examples, only ones I find are my own previous tries :D